File: frontend/src/backend/saleService.js
import {config, appFetch} from './appFetch';

export const buyTicket = (userId,bankCard,numberTickets,sessionId, onSuccess, onErrors) => {

    appFetch(`/sale/${userId}/buyTicket`, config('POST', {bankCard, numberTickets, sessionId}), onSuccess, onErrors);
}
export const deliverTicket =(saleId,bankCard,onSuccess,onErrors) =>{
    appFetch('/sale/deliverTicket',config('POST',{saleId,bankCard}),onSuccess,onErrors);

}
export const findSales = ({page},onSuccess) =>
appFetch(`/sale/tickets?page=${page}`,config('GET'),onSuccess);

File: backend/src/main/java/es/udc/paproject/backend/model/services/SaleService.java
package es.udc.paproject.backend.model.services;

import es.udc.paproject.backend.model.entities.Sale;
import es.udc.paproject.backend.model.entities.Session;
import es.udc.paproject.backend.model.entities.User;
import es.udc.paproject.backend.model.exceptions.*;

import java.util.List;
import java.util.Optional;

public interface SaleService {
    Sale buyTicket(Long userId, String bankCard, short numberTickets, Long sessionId)throws InstanceNotFoundException, NotEnoughSeatsException, TooLateException;//Usar ids e añadir excepcion capacidad, non existe, xa pasou.
    Block<Sale> findSales(Long userId, int page, int size);
    Sale deliverTicket(Long id, String bankCard)throws InstanceNotFoundException, CardDoesNotMatchException,TooLateException, AlreadyDeliveredException;//Non existe / tarxeta non coincide coa compra / session xa empezou /xa están entregadas
}
